File: netflix-conductor/simple-workers/src/main/scala/com/stulsoft/poc/simple/worker/Worker1.scala
/*
 * Copyright (c) 2019. Yuriy Stul
 */

package com.stulsoft.poc.simple.worker

import java.util

import com.netflix.conductor.client.worker.Worker
import com.netflix.conductor.common.metadata.tasks.TaskResult.Status
import com.netflix.conductor.common.metadata.tasks.{Task, TaskResult}
import com.typesafe.scalalogging.LazyLogging

/**
 * @author Yuriy Stul
 */
class Worker1(taskDefName: String) extends Worker with LazyLogging {
  override def getTaskDefName: String = taskDefName

  override def execute(task: Task): TaskResult = {
    //    logger.info(s"Executing task: $task")

    logger.debug(s"task.getInputData: ${task.getInputData}")
    task.getInputData.forEach((k, v) => logger.info(s"Key: $k, Value: $v"))
    val eventData: java.util.Map[String, Any] = task.getInputData.get("eventData").asInstanceOf[java.util.Map[String, Any]]
    val result = new TaskResult(task)
    logger.debug(s"eventData: $eventData")
    if (eventData != null) {
      eventData.forEach((k, v) => logger.debug(s"$k -> $v"))

      //Register the output of the task
      val resultData = new util.HashMap[String, Any]()
      resultData.put("state", "the state value")
      resultData.put("skipped", "no")
      resultData.put("result", "the result")
      result.getOutputData.put("resultData", resultData)

      result.setStatus(Status.COMPLETED)
    }else{
      result.setStatus(Status.FAILED)
    }
    result
  }
}

File: netflix-conductor/conductor-err1/src/main/java/com/stulsoft/netflix/conductor/err1/Worker1.java
/*
 * Copyright (c) 2019. Yuriy Stul
 */

package com.stulsoft.netflix.conductor.err1;

import com.netflix.conductor.client.worker.Worker;
import com.netflix.conductor.common.metadata.tasks.Task;
import com.netflix.conductor.common.metadata.tasks.TaskResult;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Map;

/**
 * @author Yuriy Stul
 */
public class Worker1 implements Worker {
    private static final Logger logger = LoggerFactory.getLogger(Worker1.class);

    Worker1() {
        logger.info("Initializing Worker1");
    }

    @Override
    public String getTaskDefName() {
        return "errorTestTask1";
    }

    @Override
    public TaskResult execute(Task task) {
        logger.debug("task.getInputData: {}", task.getInputData());
        task.getInputData().forEach((k, v) -> logger.info("Key: {}, value: {}", k, v));
        Map<String, Object> eventData = task.getInputData();
        TaskResult result = new TaskResult(task);
        logger.debug("eventData: {}", eventData.toString());
        try {
            Thread.sleep(300);
        } catch (Exception ignore) {
        }

        result.setReasonForIncompletion("Test1 worker: test exception ");
        // Fails the task, repeat is allowed
        result.setStatus(TaskResult.Status.FAILED);
        // Terminates the task and workflow, no repeat allowed
//        result.setStatus(TaskResult.Status.FAILED_WITH_TERMINAL_ERROR);
//        result.setStatus(TaskResult.Status.COMPLETED);
        return result;
    }
}
